### PR TITLE
[M1553] Validate the regex before making filtering

### PR DESCRIPTION
**Pull request by gridcell**
Checks the search regex string to see if it compiles before filtering sample unit methods and if fails raises a validation error rather than a 500 error

### CODE REVIEW

**Comment by gridcell** [comment]
[Invalid regular expression 500](https://trello.com/c/meLKBwBm/1553-invalid-regular-expression-500)

File: src/api/resources/sample_units/sample_unit_methods.py [hunk 1]
```diff
@@ -43,6 +43,11 @@ class SearchNonFieldFilter(django_filters.Filter):
     def filter(self, qs, value):
         value = value or ""
 
+        try:
+            re.compile(value)
+        except re.error:
+            raise exceptions.ValidationError("Invalid search")
```
Comment (saanobhaai) [inline src/api/resources/sample_units/sample_unit_methods.py:49]: So this exception will be handled and passed to response, and not cause a 500?

**Review by saanobhaai — CHANGES_REQUESTED** [review]

File: src/api/resources/sample_units/sample_unit_methods.py (hunk 1 above)
Comment (gridcell) [reply src/api/resources/sample_units/sample_unit_methods.py:49]: Yeah causes a 400 error now

File: src/api/resources/sample_units/sample_unit_methods.py [hunk 2]
```diff
@@ -163,6 +168,7 @@ class SampleUnitMethodView(BaseProjectApiViewSet):
     )
 
     def get_queryset(self):
+        print(self.request.query_params)
```
[the remark this answers is not in the document]
Comment (gridcell) [reply src/api/resources/sample_units/sample_unit_methods.py]: removed

**Pull request merged**